lctrie_ip.c

```diff
@@ -66,7 +66,7 @@ size_t subnet_dedup(lct_subnet_t *subnets, size_t size) {
   uint32_t prefix;
   size_t ndup = 0;
 
-  for (int i = 0, j = 1; j < size; ++i, ++j) {
+  for (int i = 0, j = 1; j < size; ++j) {
     // we have a duplicate!
     if (!subnet_cmp(&subnets[i], &subnets[j])) {
```

Comment:
The real fix is much simpler.  Change this from an if to a while (...) {...} that also &&'s the for loop exit condition.